File: Android/App/app/src/main/java/com/efalco777/crud_products/data/product/mapper/DataToDomainProductMapper.kt
package com.efalco777.crud_products.data.product.mapper

import com.efalco777.crud_products.data.product.model.ProductDTO
import com.efalco777.crud_products.domain.product.model.Product

class DataToDomainProductMapper {
    companion object {
        fun map(model: ProductDTO): Product {
            return Product(
                id = model.id,
                title = model.title,
                description = model.description,
                category = model.category,
                price = model.price
            )
        }
    }

}

File: Flutter/App/lib/data/product/mapper/data_to_domain_product_mapper.dart
import 'package:products_crud/data/product/model/product_dto.dart';
import 'package:products_crud/domain/product/model/product.dart';

extension ProductMapper on ProductDTO {
  Product toDomain() {
    return Product(
      id: id,
      title: title,
      description: description,
      price: price,
      category: category,
      images: images,
    );
  }
}
